### PR TITLE
docs: improve admin documentation

### PR DESCRIPTION
**Pull request by shahednasser**
- Move the admin plugin's options to the Admin Configuration documentation.
- Add a section on how to change the backend URL for both development and production.
- Fix the troubleshooting section related to port forwarding.
- General fixes.

### CODE REVIEW

File: www/apps/docs/content/admin/configuration.mdx [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+    // other scripts...
+    "dev:admin": "medusa-admin develop"
+  }
+}
+```
+
+You can add the following options to the `medusa-admin develop` command:
+
+- `--backend` or `-b`: a string specifying the URL of the Medusa backend. By default, it's the value of the environment variable `MEDUSA_ADMIN_BACKEND_URL`. For example, `medusa-admin dev --backend example.com`.
+- `--port` or `-p`: the port to run the admin on. By default, it's `7001`. For example, `medusa-admin dev --port 8000`.
```
Comment (petr0x) [inline www/apps/docs/content/admin/configuration.mdx]: medusa-admin dev is not a valid command:
error: unknown command 'dev'

Comment (shahednasser) [reply www/apps/docs/content/admin/configuration.mdx]: thanks for catching this! Seems like I missed this one. Will update it

**Review by kasperkristensen — APPROVED** [review]
LGTM!

Only comment I have, is on the note to enable autoRebuild in the various deployment guides. A good amount of these providers offer some very weak entry plans, that don't have the necessary RAM to handle running the build on the server. Might make sense to address this within the note. It requires a minimum of 2GB RAM.

**Comment by shahednasser** [comment]
@kasperkristensen is the better approach then to:

- recommend hosting separately instead;
- or only add a note about increasing the provider plan to provide at least 2GB RAM

**Comment by kasperkristensen** [comment]
@shahednasser What I usually tell people is that they could take a look at using GitHub actions to run the build before deploying their server, we do that in our staging project if you want to see an example. 

People can also opt to host separately, but while it may be free to host the admin on something like Vercel, the free plan is not allowed to be used for commercial usage, so in reality you are properly not saving money by hosting it by there.

So people have options, and I don't think one is better than the others. Maybe we should just add a small description on the different options available to people somewhere, and then link to it in a callout on all the hosting guides. In any case this is only really relevant for people who are just exploring Medusa or setting up a pet project, as most "serious" projects will have plenty of RAM to support the feature. But I still think its important to guide first timers, so this doesn't become a frustration point when testing Medusa out 😄

**Comment by shahednasser** [comment]
@kasperkristensen got it! I'm looking into improving the deployment guides as a whole, so I'll keep it out of this PR for now and add into that one. Thanks for the info!

**Review by olivermrbl — APPROVED** [review]

**Pull request merged**